src/server/object_services/object_server.js

```diff
@@ -1415,6 +1416,14 @@ async function find_object_md(req) {
         obj = await MDStore.instance().find_object_latest(req.bucket._id, req.rpc_params.key);
     }
     check_object_mode(req, obj, 'NO_SUCH_OBJECT');
+
+    // fetch the objects parts ranges if a partNumber was requested
+    // and the object was uploaded using multipart upload
+    if (req.rpc_params.part_number && obj.num_parts) {
+        const parts = await MDStore.instance().find_all_parts_of_object(obj);
```

Comment:
Why loading all parts? Why not just the requested one?

Reply:
Hey Guy. Good point. Refactored the change a bit. Added new MDStore interface to fetch a single part range for a specific part number, filtering is performed by the DB now. On metadata level, over RPC boundary, pass single requested range star/end instead of the whole object parts array.